### PR TITLE
Fixed #27257 -- Used the ->> operator when filtering  builtin text lookups on JSONField keys.

### PR DESCRIPTION
**Pull request by charettes**

### CODE REVIEW

File: tests/postgres_tests/test_json.py [hunk 1]
```diff
@@ -237,6 +238,41 @@ def test_usage_in_subquery(self):
             self.objs[7:9]
         )
 
+    def test_icontains(self):
+        self.assertFalse(
```
Comment (timgraham) [inline tests/postgres_tests/test_json.py]: I'd single line these.

File: tests/postgres_tests/test_json.py [hunk 2]
```diff
@@ -124,6 +124,7 @@ def setUpTestData(cls):
                 'k': True,
                 'l': False,
             }),
+            JSONModel.objects.create(field={'foo': 'bar'})
```
Comment (timgraham) [inline tests/postgres_tests/test_json.py]: add trailing comma

File: django/contrib/postgres/fields/jsonb.py [hunk 3]
```diff
@@ -99,14 +103,68 @@ def as_sql(self, compiler, connection):
...
+KeyTransform.register_lookup(KeyTransformIContains)
+KeyTransform.register_lookup(KeyTransformStartsWith)
+KeyTransform.register_lookup(KeyTransformIStartsWith)
+KeyTransform.register_lookup(KeyTransformEndsWith)
+KeyTransform.register_lookup(KeyTransformIEndsWith)
+KeyTransform.register_lookup(KeyTransformRegex)
+KeyTransform.register_lookup(KeyTransformIRegex)
+
+
+class KeyTextTransform(KeyTransform):
```
Comment (timgraham) [inline django/contrib/postgres/fields/jsonb.py]: Would it be clearer to declare this before it's used?

File: django/contrib/postgres/fields/jsonb.py [hunk 4]
```diff
@@ -99,14 +103,68 @@ def as_sql(self, compiler, connection):
...
             int(self.key_name)
         except ValueError:
             lookup = "'%s'" % self.key_name
         else:
             lookup = "%s" % self.key_name
-        return "(%s -> %s)" % (lhs, lookup), params
+        return "(%s %s %s)" % (lhs, self.operator, lookup), params
+
+
+class KeyTransformTextLookupMixin(object):
```
Comment (timgraham) [inline django/contrib/postgres/fields/jsonb.py]: Add a docstring or some comments about the purpose?

File: django/contrib/postgres/fields/jsonb.py (hunk 3 above)
Comment (charettes) [inline django/contrib/postgres/fields/jsonb.py]: It would.

File: django/contrib/postgres/fields/jsonb.py [hunk 5]
```diff
@@ -99,14 +103,70 @@ def as_sql(self, compiler, connection):
...
+class KeyTextTransform(KeyTransform):
+    operator = '->>'
+    nested_operator = '#>>'
+    _output_field = TextField()
+
+
+class KeyTransformTextLookupMixin(object):
+    """
+    Mixin meant to be combined with lookup expecting a text lhs from a
+    JSONField key lookup.
```
Comment (timgraham) [inline django/contrib/postgres/fields/jsonb.py]: I guess I was thinking something like example SQL would be useful.

File: django/contrib/postgres/fields/jsonb.py [hunk 6]
```diff
@@ -99,14 +103,70 @@ def as_sql(self, compiler, connection):
...
+
+class KeyTextTransform(KeyTransform):
+    operator = '->>'
+    nested_operator = '#>>'
+    _output_field = TextField()
+
+
+class KeyTransformTextLookupMixin(object):
+    """
+    Mixin meant to be combined with lookup expecting a text lhs from a
```
Comment (timgraham) [inline django/contrib/postgres/fields/jsonb.py]: "Mixin for combining with a lookup"

**Comment by timgraham** [comment]
For the commit message, maybe "Fixed builtin text lookups on JSONField keys." is a bit simpler (rather than describing implementation details)?

**Pull request merged**